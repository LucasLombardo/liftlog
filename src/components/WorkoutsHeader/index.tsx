import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
import { useTranslation } from "react-i18next";

export default function WorkoutsHeader({
  onPrev,
  onNext,
  title,
}: WorkoutsHeaderProps) {
  const { t } = useTranslation();
  return (
    <View style={s.header}>
      <TouchableOpacity
        style={s.button}
        onPress={onPrev}
        accessibilityLabel={t("workouts.previous_day")}
      >
        <Text>{"<"}</Text>
      </TouchableOpacity>
      <Text style={s.title}>{title}</Text>
      <TouchableOpacity
        style={s.button}
        accessibilityLabel={t("workouts.next_day")}
        onPress={onNext}
      >
        <Text>{">"}</Text>
      </TouchableOpacity>
    </View>
  );
}

const s = StyleSheet.create({
  header: {
    backgroundColor: "#ddd",
    alignItems: "center",
    justifyContent: "space-evenly",
    flexDirection: "row",
    borderColor: "#444",
    borderStyle: "solid",
    borderBottomWidth: 1,
  },
  button: {
    paddingVertical: 6,
    paddingHorizontal: 10,
    borderColor: "#444",
    borderStyle: "solid",
    borderWidth: 1,
    marginVertical: 2,
  },
  title: {
    minWidth: 120,
    textAlign: "center",
  },
});

interface WorkoutsHeaderProps {
  onPrev: () => void;
  onNext: () => void;
  title: string;
}
