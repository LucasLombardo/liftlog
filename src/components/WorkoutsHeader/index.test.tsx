import WorkoutsHeader from "./index";
import { render, screen, fireEvent } from "@testing-library/react-native";

describe("WorkoutsHeader", () => {
  it("component matches snapshot", () => {
    render(
      <WorkoutsHeader title="test" onPrev={() => null} onNext={() => null} />
    );
    expect(screen).toMatchSnapshot();
  });

  it("buttons trigger callbacks", () => {
    const props = {
      title: "test",
      onPrev: jest.fn(),
      onNext: jest.fn(),
    };
    render(<WorkoutsHeader {...props} />);

    const prevButton = screen.getByText("<");
    const nextButton = screen.getByText(">");

    fireEvent.press(prevButton);
    expect(props.onPrev).toHaveBeenCalledTimes(1);
    expect(props.onNext).not.toHaveBeenCalled();

    fireEvent.press(nextButton);
    expect(props.onPrev).toHaveBeenCalledTimes(1);
    expect(props.onNext).toHaveBeenCalledTimes(1);
  });
});
