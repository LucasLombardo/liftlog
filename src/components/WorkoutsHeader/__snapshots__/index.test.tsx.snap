// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutsHeader component matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ddd",
      "borderBottomWidth": 1,
      "borderColor": "#444",
      "borderStyle": "solid",
      "flexDirection": "row",
      "justifyContent": "space-evenly",
    }
  }
>
  <View
    accessibilityLabel="workouts.previous_day"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderColor": "#444",
        "borderStyle": "solid",
        "borderWidth": 1,
        "marginVertical": 2,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 6,
      }
    }
  >
    <Text>
      &lt;
    </Text>
  </View>
  <Text
    style={
      {
        "minWidth": 120,
        "textAlign": "center",
      }
    }
  >
    test
  </Text>
  <View
    accessibilityLabel="workouts.next_day"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderColor": "#444",
        "borderStyle": "solid",
        "borderWidth": 1,
        "marginVertical": 2,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 6,
      }
    }
  >
    <Text>
      &gt;
    </Text>
  </View>
</View>
`;
